test/mandelbrot.cxx

```diff
@@ -87,37 +143,50 @@ int Drawing_Area::idle() {
   if (!window()->visible()) return 0;
   if (drawn < nextline) {
     window()->make_current();
-    int yy = drawn+y()+4;
+    int yy = drawn+y()+dy;
     if (yy >= sy && yy <= sy+sh) erase_box();
-    fl_draw_image_mono(buffer+drawn*W,x()+3,yy,W,1,1,W);
+    if (use_colors)
+      fl_draw_image(buffer+drawn*W*3, x()+dx, yy, W, 1, 3);
+    else
+      fl_draw_image_mono(buffer+drawn*W, x()+dx, yy, W, 1, 1, W);
     drawn++;
     return 1;
   }
+  int linebytes = use_colors ? W*3 : W;
   if (nextline < H) {
-    if (!buffer) buffer = new uchar[W*H];
+    if (!buffer) buffer = new uchar[linebytes*H];
     double yy = Y+(H/2-nextline)*scale/W;
     double yi = yy; if (julia) yy = jY;
-    uchar *p = buffer+nextline*W;
+    uchar *p = buffer+nextline*linebytes;
     for (int xi = 0; xi < W; xi++) {
       double xx = X+(xi-W/2)*scale/W;
       double wx = xx; double wy = yi;
       if (julia) xx = jX;
       for (int i=0; ; i++) {
-        if (i >= iterations) {*p = 0; break;}
+        if (i >= iterations) {
+          *p = 0;
+          if (use_colors) {
+           *(p+1) = 0;
+           *(p+2) = 0;
+          }
+          break;
+        }
         double t = wx*wx - wy*wy + xx;
         wy = 2*wx*wy + yy;
         wx = t;
         if (wx*wx + wy*wy > 4) {
           wx = t = 1-double(i)/(1<<10);
-          if (t <= 0) t = 0; else for (i=brightness; i--;) t*=wx;
-          *p = 255-int(254*t);
+          if (t <= 0) t = 0; else for (i=(use_colors?MAX_BRIGHTNESS-2:brightness); i--;) t*=wx;
+          int c = 255 - int(254*t);
+          if (use_colors) get_color(c, *p, *(p+1), *(p+2), brightness-MAX_BRIGHTNESS/2);
+          else *p = c;
```

Comment:
... replace 179 to 182 with
```
          if (t <= 0) t = 0; else for (i=brightness; i--;) t*=wx;
          if (use_colors) {
            get_color(t, *p, *(p+1), *(p+2));
          } else {
            *p = 255 - int(254*t);
          }
```
which seems a bit more cleaned up to me and keeps the steepness of the brightness ramp as it works in grayscale

Reply:
Thanks for reviewing.
You are probably right, cleanliness is more important than fanciness (I was not satisfied with the look at minimal brightness, so I tried to improve that). But so it is more in sync with the greyscale version.
This would mean, that `DEFAULT_BRIGHTNESS_COLOR` thing could also be removed...

If you are going to commit this PR, would you mind to do the required changes yourself? You can also push to this PR or use your own PR and close this one.